### PR TITLE
Graphite: Update config editor

### PR DESCRIPTION
**Pull request by shavonn**
#20491

### CODE REVIEW

**Review by ryantxu — APPROVED** [review]
LGTM

File: public/app/plugins/datasource/graphite/configuration/ConfigEditor.tsx [hunk 1]
```diff
@@ -1,20 +1,99 @@
...
+const graphiteVersions = [
+  { label: '0.9.x', value: '0.9' },
+  { label: '1.0.x', value: '1.0' },
+  { label: '1.1.x', value: '1.1' },
+];
+
+const graphiteTypes = Object.keys(GraphiteType).map((key: string) => ({
+  label: key,
+  value: (GraphiteType as any)[key],
+}));
```
Comment (stevenvachon) [inline public/app/plugins/datasource/graphite/configuration/ConfigEditor.tsx]:
```ts
const graphiteTypes = Object.entries(GraphiteType).map(([label: string, value: any]) => ({
  label,
  value,
}));
```

File: public/app/plugins/datasource/graphite/configuration/ConfigEditor.tsx [hunk 2]
```diff
@@ -1,20 +1,99 @@
...
+            <FormLabel>Type</FormLabel>
+            <Select
+              options={graphiteTypes}
+              value={graphiteTypes.find(type => type.value === options.jsonData.graphiteType)}
+              width={8}
+              onChange={onUpdateDatasourceJsonDataOptionSelect(this.props, 'graphiteType')}
+            />
+
+            <Button
+              style={{ marginLeft: '8px', marginTop: '5px' }}
```
Comment (scottlepp) [inline public/app/plugins/datasource/graphite/configuration/ConfigEditor.tsx]: I would avoid inline styling.  I believe the standard tool here for custom styling/css is emotion.

**Review by scottlepp — CHANGES_REQUESTED** [review]

File: public/app/plugins/datasource/graphite/configuration/ConfigEditor.tsx (hunk 2 above)
Comment (shavonn) [reply public/app/plugins/datasource/graphite/configuration/ConfigEditor.tsx]: This was already done in master. I am not 100% on why. Can investigate though. @scottlepp

**Comment by srclosson** [comment]
The graphite config editor seems to work just fine. I'm able to add/remove metrics, add agregations, move those aggregations around, delete aggregations, and the aggregations come up with the expected fields and required input.

File: public/app/plugins/datasource/graphite/configuration/ConfigEditor.tsx (hunk 1 above)
Comment (shavonn) [reply public/app/plugins/datasource/graphite/configuration/ConfigEditor.tsx]: @stevenvachon it's not in that format.

Comment (shavonn) [reply public/app/plugins/datasource/graphite/configuration/ConfigEditor.tsx]:
```
export enum GraphiteType {
  Default = 'default',
  Metrictank = 'metrictank',
}
```

File: public/app/plugins/datasource/graphite/configuration/ConfigEditor.tsx [hunk 3]
```diff
@@ -1,20 +1,105 @@
...
+              <Select
+                options={graphiteTypes}
+                value={graphiteTypes.find(type => type.value === options.jsonData.graphiteType)}
+                width={8}
+                onChange={onUpdateDatasourceJsonDataOptionSelect(this.props, 'graphiteType')}
+              />
+
+              <div className={styles.helpbtn}>
+                <Button
+                  style={{ marginLeft: '8px', marginTop: '5px' }}
```
Comment (scottlepp) [inline public/app/plugins/datasource/graphite/configuration/ConfigEditor.tsx]: can remove inline style now?

**Review by scottlepp — CHANGES_REQUESTED** [review]

File: public/app/plugins/datasource/graphite/configuration/ConfigEditor.tsx (hunk 1 above)
Comment (stevenvachon) [reply public/app/plugins/datasource/graphite/configuration/ConfigEditor.tsx]: That's  the same as `{ Default: 'default', Metrictank: 'metrictank' }`, though, right? `Object.entries()` will convert that to:
```js
[ ['Default', 'default'], ['Metrictank', 'metrictank'] ]
```

where we can destructure to `label` and `value` (within `Array::map`), then reformat to `{ label, value }` (via `Object.fromEntries()`).

Comment (shavonn) [reply public/app/plugins/datasource/graphite/configuration/ConfigEditor.tsx]: @stevenvachon I tried it but it didn't work. That was my theory for why.

Comment (stevenvachon) [reply public/app/plugins/datasource/graphite/configuration/ConfigEditor.tsx]: Any runtime or compile errors?

**Review by scottlepp — APPROVED** [review]

**Pull request merged**